src/sessions/screen-share-session-handler.ts

```diff
@@ -69,19 +69,19 @@ export default class ScreenShareSessionHandler extends BaseSessionHandler {
       });
 
       if (!this.sdk.isGuest) {
-        throwSdkError.call(this.sdk, SdkErrorTypes.not_supported, 'Screen share sessions not supported for authenticated users');
+        throw createAndEmitSdkError.call(this.sdk, SdkErrorTypes.not_supported, 'Screen share sessions not supported for authenticated users');
       }
-
+      const sessionInfo = { sessionId: session.id, conversationId: session.conversationId };
       if (!this.sdk._config.autoConnectSessions) {
         // if autoConnectSessions is 'false' and we have a guest, throw an error
         //  guests should auto accept screen share session
         const errMsg = '`autoConnectSession` must be set to "true" for guests';
-        this.log('error', errMsg, { sessionId: session.id, conversationId: session.conversationId });
-        throw new Error(errMsg);
+        throw createAndEmitSdkError.call(this.sdk, SdkErrorTypes.session, errMsg, { sessionInfo });
```

Comment:
I think it still makes sense to have them top level props, but I'd just do something like this: 
`throw createAndEmitSdkError.call(this.sdk, SdkErrorTypes.session, errMsg, sessionInfo);`. My last comment just meant that `throw createAndEmitSdkError.call(this.sdk, SdkErrorTypes.session, errMsg, { ...sessionInfo });` is the same as doing `throw createAndEmitSdkError.call(this.sdk, SdkErrorTypes.session, errMsg, sessionInfo);` so might as well get rid of the cruft. same with the other occurrences in this file.

Reply:
Ahh, I see what you are saying. I wasn't thinking about this in terms of an sdk error's `details`. I had `logger.error` in my mind. I will change it out.